### PR TITLE
Add a dockSize hidden setting for docklike-plugin minimal height / width

### PR DESCRIPTION
**Pull request by jcornuz**
Having a go at solving #6 

First bit of C++ and first use of Github, so not sure if this is really helpful :/

### CODE REVIEW

**Comment by davekeogh** [comment]
Looks pretty good. Two things you need to do and this will be ready to go in:

Forward declare dockSize. This way if it's not defined in the settings file it won't be NULL and crash the plugin.
Just put `State<int> dockSize;` with the other declarations at the top of Settings.cpp

The check you are doing for panel orientation in Dock.cpp in onPanelOrientationChange needs to be duplicated in init.

File: src/Settings.cpp [hunk 1]
```diff
@@ -134,6 +134,12 @@ namespace Settings
 			});
 
 		gchar** pinnedListBuffer = g_key_file_get_string_list(mFile, "user", "pinned", NULL, NULL);
+		
+		dockSize.setup(g_key_file_get_integer(mFile, "user", "dockSize", NULL),
```
Comment (davekeogh) [inline src/Settings.cpp]: Move this block down beneath the pinnedAppList, so it's not in between two related pieces of code.

**Review by davekeogh — CHANGES_REQUESTED** [review]

**Comment by jcornuz** [comment]
I think that should do - hadn't realized the forward declaration stuff... 

Thanks for your help!

**Comment by davekeogh** [comment]
Thanks and congrats. Your code is in main now.

**Pull request merged**